src/Compilers/CSharp/Test/Semantic/Semantics/PrimaryConstructorTests.cs

```diff
@@ -8602,6 +8810,1964 @@
+    {
+        Test(new S1());
+        Test(new S2());
+    }
+
+    static void Test<T>(T x) where T : unmanaged {}
+}
+";
+            var comp = CreateCompilation(source, options: TestOptions.ReleaseExe);
+
+            comp.VerifyEmitDiagnostics(
+                // (7,18): warning CS8907: Parameter 'p1' is unread. Did you forget to use it to initialize the property with that name?
+                // struct S2(string p1)
+                Diagnostic(ErrorCode.WRN_UnreadRecordParameter, "p1").WithArguments("p1").WithLocation(7, 18),
+                // (15,9): error CS8377: The type 'S1' must be a non-nullable value type, along with all fields at any level of nesting, in order to use it as parameter 'T' in the generic type or method 'Program.Test<T>(T)'
+                //         Test(new S1());
+                Diagnostic(ErrorCode.ERR_UnmanagedConstraintNotSatisfied, "Test").WithArguments("Program.Test<T>(T)", "T", "S1").WithLocation(15, 9)
+                );
+
+            Assert.NotEmpty(comp.GetTypeByMetadataName("S1").InstanceConstructors.OfType<SynthesizedPrimaryConstructor>().Single().GetCapturedParameters());
+            Assert.Empty(comp.GetTypeByMetadataName("S2").InstanceConstructors.OfType<SynthesizedPrimaryConstructor>().Single().GetCapturedParameters());
+        }
+
+        [Fact]
+        public void ParameterCapturing_057_ColorColor_MemberAccess_InstanceAndStatic_Property()
+        {
+            //  public class Color
+            //  {
+            //      public int P => 1;
+            //      public static string P => "1";
+            //  }
+            var ilSource = @"
+.class public auto ansi beforefieldinit Color
+    extends System.Object
+{
+    .method public hidebysig specialname rtspecialname 
+        instance void .ctor () cil managed 
+    {
+        .maxstack 8
+
+        IL_0000: ldarg.0
+        IL_0001: call instance void System.Object::.ctor()
+        IL_0006: ret
+    }
+
+    .method public hidebysig specialname 
+        instance int32 get_P () cil managed 
+    {
+        .maxstack 8
+
+        IL_0000: ldc.i4.1
+        IL_0001: ret
+    }
+
+    .property instance int32 P()
+    {
+        .get instance int32 Color::get_P()
+    }
+
+    .method public hidebysig specialname static 
+        string get_P () cil managed 
+    {
+        .maxstack 8
+
+        IL_0000: ldstr ""1""
+        IL_0005: ret
+    }
+
+    .property string P()
+    {
+        .get string Color::get_P()
+    }
+}
+";
+
+            var source = @"
+struct S1(Color Color)
+{
+    public void Test()
+    {
+        _ = Color.P;
+    }
+}
+";
+            var comp = CreateCompilationWithIL(source, ilSource, options: TestOptions.ReleaseDll);
+
+            comp.VerifyEmitDiagnostics(
+                // (6,19): error CS0229: Ambiguity between 'Color.P' and 'Color.P'
+                //         _ = Color.P;
+                Diagnostic(ErrorCode.ERR_AmbigMember, "P").WithArguments("Color.P", "Color.P").WithLocation(6, 19)
```

Comment:
>ERR_AmbigMember [](http://example.com/codeflow?start=37&length=15)

I'm curious why this also isn't `ERR_AmbiguousPrimaryConstructorParameterAsColorColorReceiver`?

Reply:
> I'm curious why this also isn't ```ERR_AmbiguousPrimaryConstructorParameterAsColorColorReceiver```?

Because lookup fails with the ```ERR_AmbigMember```, C# doesn't support overloading of regular properties. The ```ERR_AmbiguousPrimaryConstructorParameterAsColorColorReceiver``` is reported if lookup succeeds